src/main/resources/db/migrations/V001__CreateSchemaTables.sql

```diff
@@ -0,0 +1,228 @@
+    name VARCHAR(255) NOT NULL,
+    location_code VARCHAR(25)
+);
+
+CREATE TABLE public.rooms (
+    id UUID PRIMARY KEY,
+    room VARCHAR(45) NOT NULL
+);
+
+CREATE TABLE public.court_region (
+    id UUID PRIMARY KEY,
+    court_id UUID REFERENCES courts(id) NOT NULL,
+    region_id UUID REFERENCES regions(id) NOT NULL
+);
+
+CREATE TABLE public.courtrooms (
+    id UUID PRIMARY KEY,
+    court_id UUID REFERENCES courts(id) NOT NULL,
+    room_id UUID REFERENCES rooms(id) NOT NULL
+);
+
+CREATE TABLE public.cases (
+    id UUID PRIMARY KEY,
+    court_id UUID REFERENCES courts(id) NOT NULL ,
+    reference VARCHAR(25) NOT NULL, 
+    test BOOLEAN DEFAULT FALSE,
+    deleted_at TIMESTAMPTZ DEFAULT NULL,
+    created_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),
+    modified_at TIMESTAMPTZ DEFAULT NOW()
+);
+
+
+CREATE TABLE public.participants (
+    id UUID PRIMARY KEY ,
+    case_id UUID REFERENCES cases(id) NOT NULL,
+    participant_type PARTICIPANT_TYPE NOT NULL,
+    first_name VARCHAR(100) NOT NULL,
+    last_name VARCHAR(100) NOT NULL,
+    deleted_at TIMESTAMPTZ DEFAULT NULL,
+    created_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),
+    modified_at TIMESTAMPTZ DEFAULT NOW()
+);
+
+CREATE TABLE public.bookings (
+    id UUID PRIMARY KEY,
+    case_id UUID REFERENCES cases(id) NOT NULL,
+    scheduled_for TIMESTAMPTZ NOT NULL,
+    deleted_at TIMESTAMPTZ DEFAULT NULL,
+    created_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),
+    modified_at TIMESTAMPTZ DEFAULT NOW()
+);
+
+
+CREATE TABLE public.booking_participant (
+    id UUID PRIMARY KEY,
+    participant_id UUID REFERENCES participants(id) NOT NULL,
+    booking_id UUID REFERENCES bookings(id) NOT NULL  
+);
+
+CREATE TABLE public.users (
+    id UUID PRIMARY KEY,
+    first_name VARCHAR(100) NOT NULL,
+    last_name VARCHAR(100) NOT NULL,
+    email VARCHAR(100) NOT NULL,
+    organisation VARCHAR(250),
+    phone VARCHAR(50),
+    deleted_at TIMESTAMPTZ DEFAULT NULL,
+    created_at TIMESTAMPTZ DEFAULT NOW(),
+    modified_at TIMESTAMPTZ DEFAULT NOW()
+);
+
+CREATE TABLE public.roles (
+    id UUID PRIMARY KEY,
+    name VARCHAR(45) NOT NULL
+);
+
+CREATE TABLE public.permissions (
+    id UUID PRIMARY KEY,
+    name VARCHAR(45) NOT NULL
+);
+
+CREATE TABLE public.role_permission (
+    id UUID PRIMARY KEY,
+    role_id UUID REFERENCES roles(id),
+    permission_id UUID REFERENCES permissions(id)
+);
+
+
+CREATE TABLE public.app_access (
+    id UUID PRIMARY KEY,
+    user_id UUID REFERENCES users(id) NOT NULL,
+    court_id UUID REFERENCES courts(id) NOT NULL,
+    role_id UUID REFERENCES roles(id) NOT NULL,
+    last_access TIMESTAMPTZ,
+    active BOOLEAN DEFAULT TRUE,
+    deleted_at TIMESTAMPTZ DEFAULT NULL,
+    created_at TIMESTAMPTZ DEFAULT NOW(),
+    modified_at TIMESTAMPTZ DEFAULT NOW()
+);
+
+
+
+CREATE TABLE public.capture_sessions (
+    id UUID PRIMARY KEY,
+    booking_id UUID REFERENCES bookings(id) NOT NULL,
+    parent_recording_id UUID NOT NULL,
```

Comment:
Either `capture_session.parent_recording_id` or `recordings.capture_session_id` needs to be nullable because...
- You cannot create a capture session without having a recording (since `parent_recording_id NOT NULL`)
- BUT, you cannot create a recording without having a capture session (since `capture_session_id NOT NULL`)

Reply:
`parent_recording_id` should belong in the `recordings` table. it should be null if it is the first recording. then any edits that are made from another recording will have the `parent_recording_id` set.

Reply:
```suggestion

```